lib/pos_transaction_register.go

```diff
@@ -249,6 +250,18 @@ func (tr *TransactionRegister) GetFeeTimeTransactions() []*MempoolTx {
 	return txns
 }
 
+// GetTransaction returns the transaction with the given hash if it exists in the register, or nil otherwise.
+func (tr *TransactionRegister) GetTransaction(hash *BlockHash) *MempoolTx {
+	tr.RLock()
+	defer tr.RUnlock()
+
+	if hash == nil {
+		return nil
+	}
```

Comment:
```suggestion
	if hash == nil {
		return nil
	}
	tr.RLock()
	defer tr.RUnlock()
```
We can acquire the read lock after handling the nil case. This way we won't have to wait for writes in the nil block hash case.